File: FrogRiverOne.py
# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"
from sets import Set
def solution(X, A):
    # write your code in Python 2.7
    if X > len(A):
        return -1
    SetA=Set()
    ret = -1
    for i,k in enumerate(A):
        if k<=X and k not in SetA:
            SetA.add(k)
        if len(SetA) == X:
            ret = i
            break
    return ret

File: Counting Elements/FrogRiverOne.cpp
// you can use includes, for example:
// #include <algorithm>
#include <set>
// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(int X, vector<int> &A) {
    // write your code in C++11 (g++ 4.8.2)
    if (X < 0 || X > int(A.size()) || A.size() == 0)
    {
        return -1;
    }
    set<int> B{};
    
    int ret = -1;
    for (int i = 0; i < int(A.size()); i++)
    {
        if (A[i]>0 && A[i]<= X)
        {
            B.insert(A[i]);
        }
        if (int(B.size()) == X)
        {
            ret = i;
            break;
        }
    }
    return ret;
}